### PR TITLE
Suggestions to refine embedded browser

### PR DESCRIPTION
**Pull request by damithc**
Some TODOs added. @yl-coder to continue the branch.

### CODE REVIEW

**Comment by yl-coder** [comment]
Noted

File: .$TEMP_ADDRESS_BOOK [hunk 1]
```diff
@@ -1,2 +1,73 @@
...
+        <postalCode></postalCode>
+        <street></street>
+    </persons>
+    <tags>
+        <name>relatives</name>
+    </tags>
+    <tags>
+        <name>friends</name>
+    </tags>
+</addressbook>
```
Comment (damithc) [inline .$TEMP_ADDRESS_BOOK]: Commited by mistake?

Comment (yl-coder) [inline .$TEMP_ADDRESS_BOOK]: Yes. Fixed.

**Comment by yl-coder** [comment]
@damithc Any more suggestion and comments before i merged this PR?

File: src/main/java/hubturbo/browser/HyperBrowser.java [hunk 2]
```diff
@@ -122,6 +120,33 @@ public synchronized void clearPage(URL url) {
...
+                return false;
+            }
+        }).findAny();
+
+        if (page.isPresent()) {
+            page.get().getBrowser().loadHTML(htmlCode);
+            replaceBrowserView(page.get().getBrowser().getBrowserView());
+            return page.get();
+        }
+        else {
```
Comment (damithc) [inline src/main/java/hubturbo/browser/HyperBrowser.java]: coding style

File: src/test/java/address/unittests/browser/HyperBrowserTest.java [hunk 3]
```diff
@@ -42,14 +43,14 @@ public HyperBrowserTest() throws MalformedURLException {
...
-        HyperBrowser browser = new HyperBrowser(HyperBrowser.Type.FULL_FEATURE_BROWSER, 1, Optional.empty());
+        HyperBrowser browser = new HyperBrowser(new EmbeddedBrowserFactory(EmbeddedBrowserFactory.Type.FULL_FEATURE_BROWSER), 1, Optional.empty());
         Page page = browser.loadUrl(new URL("https://github.com"));
         assertTrue(UrlUtil.compareBaseUrls(page.getBrowser().getOriginUrl(), new URL("https://github.com")));
     }
 
     @Test
     public void testFullFeatureBrowser_loadUrls_urlsAssigned() throws MalformedURLException, IllegalAccessException, NoSuchFieldException, InterruptedException {
-        HyperBrowser browser = new HyperBrowser(HyperBrowser.Type.FULL_FEATURE_BROWSER, 3, Optional.empty());
+        HyperBrowser browser = new HyperBrowser(new EmbeddedBrowserFactory(EmbeddedBrowserFactory.Type.FULL_FEATURE_BROWSER), 3, Optional.empty());
```
Comment (damithc) [inline src/test/java/address/unittests/browser/HyperBrowserTest.java]: It should be like this?:
```
HyperBrowser browser = EmbeddedBrowserFactory.create(EmbeddedBrowserFactory.Type.FULL_FEATURE_BROWSER);
```

Comment (damithc) [inline src/test/java/address/unittests/browser/HyperBrowserTest.java]: Look up 'Factory pattern' for examples.

Comment (yl-coder) [inline src/test/java/address/unittests/browser/HyperBrowserTest.java]: We are trying to abstract away the concrete implementation of EmbeddedBrowser and not HyperBrowser?

**Comment by yl-coder** [comment]
Ready for review

File: src/main/java/hubturbo/embeddedbrowser/HyperBrowser.java [hunk 4]
```diff
@@ -122,6 +102,33 @@ public synchronized void clearPage(URL url) {
...
+                return false;
+            }
+        }).findAny();
+
+        if (page.isPresent()) {
+            page.get().getBrowser().loadHTML(htmlCode);
+            replaceBrowserView(page.get().getBrowser().getBrowserView());
+            return page.get();
+        }
+        else {
```
Comment (damithc) [inline src/main/java/hubturbo/embeddedbrowser/HyperBrowser.java]: Why else is in a new line?

**Comment by damithc** [comment]
Had a quick look. No further comments.

File: src/main/java/hubturbo/embeddedbrowser/HyperBrowser.java (hunk 4 above)
Comment (yl-coder) [inline src/main/java/hubturbo/embeddedbrowser/HyperBrowser.java]: mistake, fixed

**Pull request merged**